odbc_fdw.c

```diff
@@ -1116,8 +1116,8 @@ Datum odbc_tables_list(PG_FUNCTION_ARGS)
 		datafctx->currentRow = currentRow;
 		SRF_RETURN_NEXT(funcctx, result);
 	} else {
-		odbc_disconnection(&env, &dbc);
 		SRF_RETURN_DONE(funcctx);
+		odbc_disconnection(&env, &dbc);
```

Comment:
This will never be called since `SRF_RETURN_DONE` will return from the function.

Reply:
Fixed. This function uses SRF and I was trying to disconnect it using wrong handlers. I stored these handlers into the `user_fctx` and using them to disconnect.